symbols/overlay29.yml

```diff
@@ -8536,6 +8536,102 @@ overlay29:
+        return: bool
+    - name: CheckVariousConditions
+      address:
+        EU: 0x230156C
+      description: |-
+        Returns 0 if none of these conditions holds for the given entity: is a rescue client, 
+        doesn’t gain experience (a mission client/story teammate?), is a terrified non-team-leader, 
+        meets any of the conditions in CheckVariousStatuses2 (with blind_check = 0), is charging a two-turn move.
+
+        r0: Entity pointer
+        return: bool
+    - name: MonsterCanThrowItems
+      address:
+        EU: 0x2302D94
+      description: |-
+        Returns a boolean indicating whether or not the given monster can throw items based on its monster id.
+
+        r0: Monster pointer
+        return: bool
+    - name: GetPossibleAiArcItemTargets
+      address:
+        EU: 0x230FAA0
+      description: |-
+        Gets the positions of all targets that an AI can hit with an item thrown in an arc, such as a Gravelerock.
+        The number of positions in the array will be stored in AI_THROWN_ITEM_ACTION_CHOICE_COUNT.
+
+        r0: Entity pointer
+        r1: Item struct pointer
+        r2: [output] Array of size 20 for storing position structs
+        r3: If false, will roll GetAiUseItemProbability every time a position is checked and not add it if it rolls false.
+    - name: GetPossibleAiThrownItemDirections
+      address:
+        EU: 0x230F824
+      description: |-
+        If the entity can throw an item at a target in a certain direction, adds that direction to THROWN_ITEM_DIRECTIONS and the probability of throwing it to THROWN_ITEM_PROBABILITIES.
+        The size of the arrays will be stored in AI_THROWN_ITEM_ACTION_CHOICE_COUNT.
+        The caller function will select the direction to throw the item by iterating through the array(s), rolling the probability, and then throwing in that direction if the roll succeeds.
+        Nothing will be thrown if all rolls fail.
+
+        r0: Entity pointer
+        r1: Integer in {1, 2}. If 1, target allies; if 2, target enemies.
+        r2: Item struct pointer
+        r3: If false, will call GetAiUseItemProbability to get the probability of throwing in a certain direction.
+        If true, the added probability will always be 100.
+    - name: SetDecoyTracker
+      address:
+        EU: 0x22FBF44
+      description: |-
+        If there is a decoy on the floor that the entity can see, sets the entity's decoy_ai_tracker to 1 or 2, 
+        depending on whether the attacker who caused the decoy status is on the team or not, respectively.
+
+        r0: Entity pointer
+    - name: AiDecideUseItem
+      address:
+        EU: 0x230F388
+      description: |-
+        Decides whether or not an AI should use its held item and updates its action_data fields accordingly.
+
+        r0: Entity pointer
+    - name: TestItemAiFlag
```

Comment:
Can we add a function prototype?

Reply:
This is a function in arm9 that already has a header, I put this in overlay29 by mistake.